packages/framework/src/Model/Product/ProductDataFactory.php

```diff
@@ -118,18 +127,27 @@ public function createDefault()
         $productData->descriptions = $nullForAllDomains;
         $productData->shortDescriptions = $nullForAllDomains;
         $productData->accessories = [];
-
-        return $productData;
     }
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
      * @return \Shopsys\FrameworkBundle\Model\Product\ProductData
      */
-    public function createFromProduct(Product $product)
+    public function createFromProduct(Product $product): ProductData
     {
-        $productData = $this->createDefault();
+        $productData = new ProductData();
+        $this->fillNew($productData);
```

Comment:
why? is this necessary? You do not do that anywhere else

Reply:
btw, generally, I was thinking, why `createFrom*` methods do not use `create()` methods in factories? Would not it be better? e.g.:
```
public function createFromProduct(Product $product): ProductData
{
        $productData = $this->create();
        $this->fillFromProduct($productData, $product);

        return $productData;
}
```

Reply:
Good point.
If method `createFromProduct` depends on `create` method, and I need to add something into `create`, then I have to think how it influences the dependent method. So I decided to separate them, so we are able to add something into `create` without influencing the `createFromProduct`.

Couple years ago, I did similar approach in one project, and in the end I had troubles, because changing one method influenced the second one, I started to create new method like `createRaw`, `createDefault`, and it was... ununderstable

Reply:
I see, ok, that makes sense :+1: And what about my question on the `$this->fillNew($productData);` above?

Reply:
I thought that `fillNew` method contains something that `fillFromProduct` doesnt. But it is not true, so I'm removing the calling. Good point :100: